src/models/animal.ts

```diff
@@ -0,0 +1,51 @@
+export interface AnimalConstructor {
+    new (name: string, creator: string, code: string): Animal;
+}
+
+export class Animal {
+    name!: string;
+    column!: number;
+    row!: number;
+    score!: number;
+    error: string | undefined;
+    color!: string;
+    creator!: string;
+    doStep!: Function;
+
+    public static CreateAnimal(Ctor: AnimalConstructor, antName: string, creator: string, code: string): Animal {
+      return new Ctor(antName, creator, code);
```

Comment:
probably just being a derp and missed it, but what is a CTOR?

Reply:
constructor, it's a typescript thing that I have not fully digested yet.